File: jeong-hyuk/2월8일/하노이의탑.py
def hanoi(n, start, end, temp):
    if n == 1:
        print(start, "->", end)
    else:
        hanoi(n - 1, start, temp, end)
        print(start, "->", end)
        hanoi(n - 1, temp, end, start)

# 예시
def solution(n):
    print("하노이의 탑 이동 순서:")
    hanoi(n, 1, 3, 2)

File: yht0827/2월/2월 8일/하노이의탑.java
import java.util.*;

class Solution {

	ArrayList<int[]> list;

	public int[][] solution(int n) {
		list = new ArrayList<>();
		hanoi(1, 2, 3, n);
		int[][] result = new int[list.size()][2];
		for (int i = 0; i < list.size(); i++) {
			result[i][0] = list.get(i)[0];
			result[i][1] = list.get(i)[1];
		}
		return result;
	}

	public void hanoi(int s, int v, int e, int n) {
		int[] move = {s, e};

		if (n == 1)
			list.add(move);
		else {
			hanoi(s, e, v, n - 1);
			list.add(move);
			hanoi(v, s, e, n - 1);
		}
